index.js

```diff
@@ -56,7 +57,8 @@ module.exports.concat = function (opts, cb) {
       chunks.push(chunk)
     })
     res.on('end', function () {
-      cb(null, res, Buffer.concat(chunks))
+      var data = Buffer.concat(chunks)
+      cb(null, res, opts.json ? JSON.parse(data) : data)
```

Comment:
There are two issues here:

1. `JSON.parse` only support `Buffer` arguments in Node. In the browser, this will throw an exception. So, that should be `JSON.parse(data.toString())`.

2. `JSON.parse` will throw an exception when it's called with an invalid JSON string. So if the server returns an HTML 404 page, this will just throw an exception and the user's callback will never be called. `JSON.parse` should always be wrapped in a try-catch 👍 

When there's an error parsing the data as JSON, we should call `cb` with the `err` from the `catch` block.